src/jahongir/service/CategoryService.java

```diff
@@ -15,43 +15,58 @@ public Category[] getCategories() {
     @Override
     public boolean add(Object object) {
         Category category = (Category) object;
-
-        if (validate(category)) {}
-
-        return false;
+        if (validate(category)) {
+            categories[indexCategories++] = category;
+            return true;
+        } else {
+            return false;
+        }
     }
 
     @Override
     public boolean delete(UUID id) {
+
         return false;
     }
 
     @Override
     public boolean update(UUID id, Object object) {
         Category updateCategory = (Category) object;
-
         for (int i = 0; i < indexCategories; i++) {
             if (categories[i].getId().equals(id)) {
                 categories[i].setName(updateCategory.getName());
                 return true;
             }
         }
-
         return false;
     }
 
     @Override
     public Object[] list(UUID id) {
-        Category[] parentCategories = new Category[100];
-        int index = 0;
-
-        return parentCategories;
+        Category[] parentCategories = new Category[indexCategories];
+        int count = 0;
+        for (int i = 0; i < indexCategories; i++) {
+            if (categories[i].getId() != null && categories[i].getId().equals(id)) {
+                parentCategories[count++] = categories[i];
+            }
+        }
+        Category[] result = new Category[count];
+        for (int i = 0; i < count; i++) {
+            result[i] = parentCategories[i];
+        }
+        return result;
     }
 
     private boolean validate(Category category) {
```

Comment:
you should add isExist private method to check object's name is exist or not. then you can use this method in your validate method

isExist dgan  method yozishiz kere object ni bormi yomi tekshirish uchun. misol uchun Category da shunaqa Category name bormi yomi tekshirish uchun. chunki agar shunaqa Category bosa nima kere yana qoshib. validatsiyadan otmasigi kere shunaqa name bosa